src/fges.jl

```diff
@@ -0,0 +1,528 @@
+####################################################################
+# Data Structures
+####################################################################
+
+#Collection of variables to pass along to different functions in the FGES algorithm.  
+struct ParseData{M, F<:AbstractFloat}
+    data::M #orginal data
+    normAugData::M #standardized by the mean and std of each column, appended with ones column at end
+    numFeatures::Int #number of columns
+    numObservations::Int #number of rows
+    penalty::F
+
+    function ParseData(data::M, penalty) where M 
+
+        #Determine the data type of the data inputted
+        baseType = eltype(data)
+
+        #Get the dimensions of the input data
+        numObservations, numFeatures = size(data)
+
+        #Copy the data and standardize each column
+        normAugData = copy(data)
+        normAugData .-= mean(normAugData, dims=1) #subtract mean
+        normAugData ./= std(normAugData, dims=1) #divide by standard deviation
+
+        #Augment a column of ones on the end for linear regression
+        normAugData = [normAugData ones(baseType, numObservations)]
+
+        #Ensure that the penalty is the same type as the data (e.g. Float32)
+        penalty = baseType(penalty)
+
+        return new{M, baseType}(data, normAugData, numFeatures, numObservations, penalty)
+    end
+end  
+
+#Simple structure to hold the current edge, a subset of neighbors, and a score change
+Base.@kwdef mutable struct Step{A,B}
```

Comment:
```suggestion
Base.@kwdef mutable struct FESStep{A,B}
```
This should have a more specific name